### PR TITLE
Feat/templates

### PR DESCRIPTION
**Pull request by anngvu**
Work in progress. Followup to #23.

- Mostly #20
- Plus a couple of small updates to funs

Other notes: 
I didn't think that anyone would want to test the complementary data warehouse data prep template, because 
1) it's an onerous process and 
2) more importantly, it's going to be obsolete very soon with certain changes. 

That template is there mostly for context. However, I think this data prep template is somewhat easier to go through if interested and will be useful for at least another 6-12* months.

*depends on progress of data mesh to bring Google Analytics data into Snowflake, etc.

### CODE REVIEW

File: inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+- We'll take the largest number of users seen for any project page and approximate that as maximum number of users for that project.
+
+```{r, eval=params$eval}
+
+ga_data <- rbindlist(ga_data, idcol = "projectId")
+ga_data <- ga_data[, .(sum_pageviews = sum(screenPageViews), max_users = max(totalUsers)), by = "projectId"]
+```
+
+We augment it with release groupings data. 
+We can check expected differences in views between released and non-released studies, given that non-released only have direct contributor/collaborators, while released should have additional community users.
```
Comment (jaybee84) [inline inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd:369]: As I was reading this, one thing occurred to me. Given that metadata tables may be viewable for "under-embargo" studies, and the fact that table views are considered "downloads" in the data warehouse, both the "view" from GA and "download" metrics from the warehouse for "under-embargo" studies will be affected by this behavior.

File: R/palette.R [hunk 2]
```diff
@@ -16,11 +16,13 @@ default_palette <- function() {
 
 #' All-purpose Synapse palette
 #'
-#' Return max of 10 synapse colors.
+#' Return max of 15 synapse colors.
 #' @export
 syn_palette <- function() {
```
Comment (jaybee84) [inline R/palette.R:21]: love this implementation of palette :) 
Some time ago there was a concerted effort to consolidate Sage branding colors. So adding this for consideration: https://sage-bionetworks.github.io/sagethemes/index.html

**Review by jaybee84 — COMMENTED** [review]
Wonderfully done.. Added some comments for consideration .. otherwise LGTM.

File: R/palette.R (hunk 2 above)
Comment (anngvu) [reply R/palette.R:21]: Ha, thanks, didn't know about this, good to chat with Abby about it some time, maybe.

File: inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+inferred <- unique(dw_data[id %in% not_in_fileview, .(id = paste0("syn", id), name = NAME, projectId = project)])
+
+assign_resourceType <- function(x) {
+  if(grepl("jpg|cram|crai", x)) return("experimentalData")
+  if(grepl("png", x)) return(NA_character_) # this one file does not look like experimentalData
+}
+
+assign_datatype <- function(x) {
+  if(grepl("jpg$|png$", x)) return("image")
+  if(grepl("cram$", x)) return("AlignedReads")
```
[the remark this answers is not in the document]
Comment (anngvu) [reply inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd:173]: Yeah, this is imperfect inference example.

File: inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd (hunk 1 above)
Comment (anngvu) [reply inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd:369]: May just need to clarify in main report, I think we already know that "downloads" are a little inflated.

**Pull request merged**